Add hooks to the hiddenRows and hiddenColumns plugins.

jansiegel opened the pull request:
Add the following hooks:
- beforeHideRows
- afterHideRows
- beforeUnhideRows
- afterUnhideRows
- beforeHideColumns
- afterHideColumns
- beforeUnhideColumns
- afterUnhideColumns
 #163

jansiegel committed:
Add the following hooks:
- beforeHideRows
- afterHideRows
- beforeUnhideRows
- afterUnhideRows
- beforeHideColumns
- afterHideColumns
- beforeUnhideColumns
- afterUnhideColumns
 #163

AMBudnik commented:
I'm so happy to see this PR @jansiegel !

wszymanski commented:
Please add extra test related to not calling `after...` hook when `before...` hook return `false`.

jansiegel committed:
Add tests to make sure the `after` hiding hooks don’t fire, if the `before` hooks returned `false`. #163

jansiegel committed:
Extend the functionality of the hooks in hiddenRows/hiddenColumns. #163

jansiegel committed:
- Make the plugin still run the 'after' hooks for hiding/unhiding rows/columns, if the validation was unsuccessful
- Add an additional parameter to the 'after' hooks, telling if the validation was successful
 #163

jansiegel committed:
Rename the validation function. #163

wszymanski commented on src/plugins/hiddenColumns/hiddenColumns.js:
```diff
@@ -220,6 +255,17 @@ class HiddenColumns extends BasePlugin {
+   * @param {Array} columns Array of column indexes.
+   */
+  isColumnDataValid(columns) {
+    const outOfBounds = columns.find(column => (column < 0 || column > this.hot.countCols() - 1));
```
You may use also [every](https://developer.mozilla.org/pl/docs/Web/JavaScript/Referencje/Obiekty/Array/every), but decission about using it is up to you.

jansiegel committed:
Change the use of 'find' in favor of 'every'. #163

wszymanski reviewed (approved):
<3

jansiegel commented:
Moved to https://github.com/handsontable/handsontable/pull/5651
